docx_editor/ooxml/unpack.py

```
@@ -29,10 +53,25 @@ def unpack_document(input_file: str | Path, output_dir: str | Path) -> str:
     if not input_path.exists():
         raise DocumentNotFoundError(f"Document not found: {input_file}")
 
-    # Extract and format
-    output_path.mkdir(parents=True, exist_ok=True)
+    # Reject a symlinked destination so extractall cannot write through it.
+    if output_path.is_symlink():
+        raise InvalidDocumentError(f"Output directory is a symlink: {output_dir}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)